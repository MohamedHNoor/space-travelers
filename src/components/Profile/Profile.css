.profile-container {
  display: flex;
  justify-content: space-between;
  width: 85%;
  margin: 2% auto 1% auto;
}

.rockets-container {
  min-width: 48%;
}

.mission-container {
  min-width: 48%;
}

.mission-container ul {
  border: 1px solid #e3e3e3;
}

.profile-container h2 {
  font-size: 1.5rem;
  font-weight: 500;
  color: #101;
}

.no-mission,
.no-rockets {
  font-weight: 500;
  color: #555;
  margin-top: 20px;
  opacity: 0.5;
}

.item {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 10px 0;
  padding: 1.5rem;
}

.item:not(:last-child) {
  border-bottom: 1px solid #e3e3e3;
}

/* .mission-item:not(:last-child) {
  border-bottom: 1px solid #e3e3e3;
} */

/* .no-mission {
  border
} */
