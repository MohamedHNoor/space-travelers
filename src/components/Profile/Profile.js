import React from 'react';

const Profile = () => <h1>I am the profile page</h1>;

export default Profile;
