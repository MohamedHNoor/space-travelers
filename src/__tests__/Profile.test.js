import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Profile from '../components/profile/Profile';
import '@testing-library/jest-dom';

describe('Check diplay of missions and rockets on profile', () => {
  test('Check if rockets are rended', () => {
    const myRockets = render(
      <Provider store={store}>
        <BrowserRouter>
          <Profile />
        </BrowserRouter>
      </Provider>,
    );
    expect(myRockets).toMatchSnapshot();
  });

  test('Check if Missions are rended', () => {
    const myMissions = render(
      <Provider store={store}>
        <BrowserRouter>
          <Profile />
        </BrowserRouter>
      </Provider>,
    );
    expect(myMissions).toMatchSnapshot();
  });
});
