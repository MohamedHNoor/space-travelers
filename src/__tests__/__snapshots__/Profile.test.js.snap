// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check diplay of missions and rockets on profile Check if Missions are rended 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-container"
      >
        <div
          class="rockets-container"
        >
          <h2>
            My Rockets
          </h2>
          <p
            class="no-rockets"
          >
            No Rockets reserved...
          </p>
          <ul
            style="margin-top: 10px;"
          />
        </div>
        <div
          class="mission-container"
        >
          <h2>
            My Missions
          </h2>
          <ul
            style="margin-top: 10px;"
          >
            <p
              class="no-mission"
            >
              No Mission joined...
            </p>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-container"
    >
      <div
        class="rockets-container"
      >
        <h2>
          My Rockets
        </h2>
        <p
          class="no-rockets"
        >
          No Rockets reserved...
        </p>
        <ul
          style="margin-top: 10px;"
        />
      </div>
      <div
        class="mission-container"
      >
        <h2>
          My Missions
        </h2>
        <ul
          style="margin-top: 10px;"
        >
          <p
            class="no-mission"
          >
            No Mission joined...
          </p>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Check diplay of missions and rockets on profile Check if rockets are rended 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-container"
      >
        <div
          class="rockets-container"
        >
          <h2>
            My Rockets
          </h2>
          <p
            class="no-rockets"
          >
            No Rockets reserved...
          </p>
          <ul
            style="margin-top: 10px;"
          />
        </div>
        <div
          class="mission-container"
        >
          <h2>
            My Missions
          </h2>
          <ul
            style="margin-top: 10px;"
          >
            <p
              class="no-mission"
            >
              No Mission joined...
            </p>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-container"
    >
      <div
        class="rockets-container"
      >
        <h2>
          My Rockets
        </h2>
        <p
          class="no-rockets"
        >
          No Rockets reserved...
        </p>
        <ul
          style="margin-top: 10px;"
        />
      </div>
      <div
        class="mission-container"
      >
        <h2>
          My Missions
        </h2>
        <ul
          style="margin-top: 10px;"
        >
          <p
            class="no-mission"
          >
            No Mission joined...
          </p>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
